e2e/key/key.go

```diff
@@ -0,0 +1,230 @@
+			expectExit: 0,
+		},
+		{
+			name: "import private binary wrong password",
+			args: []string{"import", c.privateExportPath},
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("theWrongPassword"), // The wrong password to decrypt the key (will fail)
+				e2e.ConsoleSendLine("somethingElse"),
+				e2e.ConsoleSendLine("somethingElse"),
+			},
+			expectExit: 2,
+		},
+		{
+			name: "import private binary",
+			args: []string{"import", c.privateExportPath},
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("e2etests"), // The password to decrypt the key
+				e2e.ConsoleSendLine("e2etests"), // Then the new password
+				e2e.ConsoleSendLine("e2etests"), // Confirm the password
+			},
+			expectExit: 0,
+		},
+		{
+			name:       "import public ascii",
+			args:       []string{"import", c.publicExportASCIIPath},
+			expectExit: 0,
+		},
+		{
+			name: "import private ascii wrong password",
+			args: []string{"import", c.privateExportASCIIPath},
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("theWrongPassword"), // The wrong password to decrypt the key (will fail)
+				e2e.ConsoleSendLine("somethingElse"),
+				e2e.ConsoleSendLine("somethingElse"),
+			},
+			expectExit: 2,
+		},
+		{
+			name: "import private ascii",
+			args: []string{"import", c.privateExportASCIIPath},
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("e2etests"), // The password to decrypt the key
+				e2e.ConsoleSendLine("e2etests"), // Then the new password
+				e2e.ConsoleSendLine("e2etests"), // Confirm the password
+			},
+			expectExit: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		c.singularityResetKeyring(t) // Remove the tmp keyring before each import
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithCommand("key"),
+			e2e.WithArgs(tt.args...),
+			e2e.ConsoleRun(tt.consoleOps...),
+			e2e.ExpectExit(tt.expectExit),
+		)
+	}
+}
+
+func (c *ctx) singularityResetKeyring(t *testing.T) {
+	// TODO: run this as non-root
+	err := os.RemoveAll(c.keyRing)
+	if os.IsNotExist(err) && err != nil {
+		t.Fatalf("unable to remove tmp keyring directory: %s", err)
+	}
+}
+
+// Run the 'key' tests in order
+func (c *ctx) singularityKeyCmd(t *testing.T) {
+	c.singularityKeyNewpair(t)
+	c.singularityKeyExport(t)
+	c.singularityKeyImport(t)
+	c.singularityKeyExport(t)
+	c.singularityKeyImport(t)
+	c.singularityKeyList(t)
+}
+
+// RunE2ETests is the main func to trigger the test suite
+func RunE2ETests(env e2e.TestEnv) func(*testing.T) {
+	c := &ctx{
+		env:                    env,
+		publicExportPath:       filepath.Join(env.TestDir, "public_key.asc"),
+		publicExportASCIIPath:  filepath.Join(env.TestDir, "public_ascii_key.asc"),
+		privateExportPath:      filepath.Join(env.TestDir, "private_key.asc"),
+		privateExportASCIIPath: filepath.Join(env.TestDir, "private_ascii_key.asc"),
+		keyRing:                filepath.Join(env.TestDir, "sypgp-test-keyring"),
+	}
+
```

Comment:
```suggestion

// todo: setting the environment variable here is a temporary solution until we
// have the option of creating a temporary directory on a per-test basis.
// When we will have such an option, we should remove the code setting `SINGULARITY_SYPGPDIR`.
```

Reply:
May not be necessary anymore? https://github.com/sylabs/singularity/pull/4144 is merged.